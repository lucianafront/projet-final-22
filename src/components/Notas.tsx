import * as React from "react";
import { NotasContext } from "../context/notas.context";
import { INotas, NotasContextType } from "../@types/notas.d";

import { FaTrashCan } from "react-icons/fa6";
import { FaRotate } from "react-icons/fa6";

const Notas = () => {
  
  const {
    editandoNota,
    deletarNota,
    notas,
    notasFiltradas,
    setNotasFiltradas,
  } = React.useContext(NotasContext) as NotasContextType;

  const filtrarNotas = (prioridade: string) => {
    console.log(prioridade);
    setNotasFiltradas(
      notas.filter((nota: INotas) => {
        if (prioridade === "") {
          return notas;
        } else {
          return nota.prioridade === prioridade;
        }
      })
    );
  };

  return (
    <div>
      <div className="circulos">
        <div className="circle todas" title="Mostrar todas as notas" onClick={() => filtrarNotas("")}></div>
        <div title="Mostar somente notas com prioridade baixa"
          className="circle baixa"
          onClick={() => filtrarNotas("baixa")}
        ></div>
        <div title="Mostar somente notas com prioridade media"
          className="circle media"
          onClick={() => filtrarNotas("media")}
        ></div>
        <div 
        title="Mostar somente notas com prioridade alta"
        className="circle alta" onClick={() => filtrarNotas("alta")}></div>
      </div>

      <ul className="note">
        {notasFiltradas.map((nota: INotas) => (
          <li key={nota.id} className={nota.prioridade} data-testid={`nota-id-${nota.id}`} >
            <h2>{nota.titulo}</h2>
            <p>{nota.conteudo}</p>

            <p style={{ paddingTop: "50px", fontSize: "0.8rem" }}>
              {nota.data.toLocaleDateString()} -- {nota.prioridade}
            </p>
            <button title="Remover nota"  onClick={() => deletarNota(nota.id)} className="delete">
              <FaTrashCan className="icones" />
            </button>
            <button title="Atualizar nota" onClick={() => editandoNota(nota.id)} className="atualiza">
              <FaRotate className="icones" />
            </button>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Notas;
