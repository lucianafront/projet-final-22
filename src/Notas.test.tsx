import React from "react";
import { render, screen } from "@testing-library/react";
import Notas from "./components/Notas";
import NotasProvider from "./context/notas.context";

describe("Notas component", () => {
  test("renders Notas component", () => {

    render(<Notas />,{ wrapper: NotasProvider})
 
    const notasElement1 = screen.getByTestId("nota-id-1");
    expect(notasElement1).toBeInTheDocument();


    const notasElement2 = screen.getByTestId("nota-id-2");
    expect(notasElement2).toBeInTheDocument();


    const notasElement3 = screen.getByTestId("nota-id-3");
    expect(notasElement3).toBeInTheDocument();
  });

  

  test("matches snapshot", () => {
      const { asFragment } =  render(<Notas />,{ wrapper: NotasProvider})
  

    expect(asFragment()).toMatchSnapshot();
  });
});
