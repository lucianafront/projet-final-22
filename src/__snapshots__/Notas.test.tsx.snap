// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notas component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="circulos"
    >
      <div
        class="circle todas"
        title="Mostrar todas as notas"
      />
      <div
        class="circle baixa"
        title="Mostar somente notas com prioridade baixa"
      />
      <div
        class="circle media"
        title="Mostar somente notas com prioridade media"
      />
      <div
        class="circle alta"
        title="Mostar somente notas com prioridade alta"
      />
    </div>
    <ul
      class="note"
    >
      <li
        class="baixa"
        data-testid="nota-id-1"
      >
        <h2>
          Nota 1
        </h2>
        <p>
          Conteudo 1
        </p>
        <p
          style="padding-top: 50px; font-size: 0.8rem;"
        >
          03/06/2024 -- baixa
        </p>
        <button
          class="delete"
          title="Remover nota"
        >
          <svg
            class="icones"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M135.2 17.7C140.6 6.8 151.7 0 163.8 0H284.2c12.1 0 23.2 6.8 28.6 17.7L320 32h96c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 96 0 81.7 0 64S14.3 32 32 32h96l7.2-14.3zM32 128H416V448c0 35.3-28.7 64-64 64H96c-35.3 0-64-28.7-64-64V128zm96 64c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16z"
            />
          </svg>
        </button>
        <button
          class="atualiza"
          title="Atualizar nota"
        >
          <svg
            class="icones"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M142.9 142.9c62.2-62.2 162.7-62.5 225.3-1L327 183c-6.9 6.9-8.9 17.2-5.2 26.2s12.5 14.8 22.2 14.8H463.5c0 0 0 0 0 0H472c13.3 0 24-10.7 24-24V72c0-9.7-5.8-18.5-14.8-22.2s-19.3-1.7-26.2 5.2L413.4 96.6c-87.6-86.5-228.7-86.2-315.8 1C73.2 122 55.6 150.7 44.8 181.4c-5.9 16.7 2.9 34.9 19.5 40.8s34.9-2.9 40.8-19.5c7.7-21.8 20.2-42.3 37.8-59.8zM16 312v7.6 .7V440c0 9.7 5.8 18.5 14.8 22.2s19.3 1.7 26.2-5.2l41.6-41.6c87.6 86.5 228.7 86.2 315.8-1c24.4-24.4 42.1-53.1 52.9-83.7c5.9-16.7-2.9-34.9-19.5-40.8s-34.9 2.9-40.8 19.5c-7.7 21.8-20.2 42.3-37.8 59.8c-62.2 62.2-162.7 62.5-225.3 1L185 329c6.9-6.9 8.9-17.2 5.2-26.2s-12.5-14.8-22.2-14.8H48.4h-.7H40c-13.3 0-24 10.7-24 24z"
            />
          </svg>
        </button>
      </li>
      <li
        class="alta"
        data-testid="nota-id-3"
      >
        <h2>
          Nota 3
        </h2>
        <p>
          Conteudo 3
        </p>
        <p
          style="padding-top: 50px; font-size: 0.8rem;"
        >
          02/06/2024 -- alta
        </p>
        <button
          class="delete"
          title="Remover nota"
        >
          <svg
            class="icones"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M135.2 17.7C140.6 6.8 151.7 0 163.8 0H284.2c12.1 0 23.2 6.8 28.6 17.7L320 32h96c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 96 0 81.7 0 64S14.3 32 32 32h96l7.2-14.3zM32 128H416V448c0 35.3-28.7 64-64 64H96c-35.3 0-64-28.7-64-64V128zm96 64c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16z"
            />
          </svg>
        </button>
        <button
          class="atualiza"
          title="Atualizar nota"
        >
          <svg
            class="icones"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M142.9 142.9c62.2-62.2 162.7-62.5 225.3-1L327 183c-6.9 6.9-8.9 17.2-5.2 26.2s12.5 14.8 22.2 14.8H463.5c0 0 0 0 0 0H472c13.3 0 24-10.7 24-24V72c0-9.7-5.8-18.5-14.8-22.2s-19.3-1.7-26.2 5.2L413.4 96.6c-87.6-86.5-228.7-86.2-315.8 1C73.2 122 55.6 150.7 44.8 181.4c-5.9 16.7 2.9 34.9 19.5 40.8s34.9-2.9 40.8-19.5c7.7-21.8 20.2-42.3 37.8-59.8zM16 312v7.6 .7V440c0 9.7 5.8 18.5 14.8 22.2s19.3 1.7 26.2-5.2l41.6-41.6c87.6 86.5 228.7 86.2 315.8-1c24.4-24.4 42.1-53.1 52.9-83.7c5.9-16.7-2.9-34.9-19.5-40.8s-34.9 2.9-40.8 19.5c-7.7 21.8-20.2 42.3-37.8 59.8c-62.2 62.2-162.7 62.5-225.3 1L185 329c6.9-6.9 8.9-17.2 5.2-26.2s-12.5-14.8-22.2-14.8H48.4h-.7H40c-13.3 0-24 10.7-24 24z"
            />
          </svg>
        </button>
      </li>
      <li
        class="media"
        data-testid="nota-id-2"
      >
        <h2>
          Nota 2
        </h2>
        <p>
          Conteudo 2
        </p>
        <p
          style="padding-top: 50px; font-size: 0.8rem;"
        >
          01/06/2024 -- media
        </p>
        <button
          class="delete"
          title="Remover nota"
        >
          <svg
            class="icones"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M135.2 17.7C140.6 6.8 151.7 0 163.8 0H284.2c12.1 0 23.2 6.8 28.6 17.7L320 32h96c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 96 0 81.7 0 64S14.3 32 32 32h96l7.2-14.3zM32 128H416V448c0 35.3-28.7 64-64 64H96c-35.3 0-64-28.7-64-64V128zm96 64c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16V432c0 8.8 7.2 16 16 16s16-7.2 16-16V208c0-8.8-7.2-16-16-16z"
            />
          </svg>
        </button>
        <button
          class="atualiza"
          title="Atualizar nota"
        >
          <svg
            class="icones"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M142.9 142.9c62.2-62.2 162.7-62.5 225.3-1L327 183c-6.9 6.9-8.9 17.2-5.2 26.2s12.5 14.8 22.2 14.8H463.5c0 0 0 0 0 0H472c13.3 0 24-10.7 24-24V72c0-9.7-5.8-18.5-14.8-22.2s-19.3-1.7-26.2 5.2L413.4 96.6c-87.6-86.5-228.7-86.2-315.8 1C73.2 122 55.6 150.7 44.8 181.4c-5.9 16.7 2.9 34.9 19.5 40.8s34.9-2.9 40.8-19.5c7.7-21.8 20.2-42.3 37.8-59.8zM16 312v7.6 .7V440c0 9.7 5.8 18.5 14.8 22.2s19.3 1.7 26.2-5.2l41.6-41.6c87.6 86.5 228.7 86.2 315.8-1c24.4-24.4 42.1-53.1 52.9-83.7c5.9-16.7-2.9-34.9-19.5-40.8s-34.9 2.9-40.8 19.5c-7.7 21.8-20.2 42.3-37.8 59.8c-62.2 62.2-162.7 62.5-225.3 1L185 329c6.9-6.9 8.9-17.2 5.2-26.2s-12.5-14.8-22.2-14.8H48.4h-.7H40c-13.3 0-24 10.7-24 24z"
            />
          </svg>
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
